tsds-grafana-handler/tsds_grafana_handler.py

```diff
@@ -81,36 +72,40 @@ def findtarget_names(tsds_result, alias_list, target_aliases):
                 if is_metric(key):
                         continue
 
+                # ex. [u'sum', u'aggregate', u'values.output', u'300', u'average', u'', u'']
                 datapoint_args = map(lambda x: x.strip(), re.split('[(,)]', key))
-
-                # [u'sum', u'aggregate', u'values.output', u'300', u'average', u'', u'']
-                if not datapoint_args[0] == u'aggregate':
-                        returnname.append({'name': key, 'target': " ".join(metric_label(tsds_result))})
-                        continue
-
-                target = 'unknown'
-                if 'percentile' in key:
-                        _, measurement, seconds, aggregation, percentile, _, _ = re.split('[(,)]', key)
+                name = None
+
+                if datapoint_args[0] != 'aggregate':
+                        # If the tsds query isn't a simple aggregate
+                        # just return the datapoints name.
+                        name = key
+                elif key in datapoint_aliases and datapoint_aliases[key] != '':
```

Comment:
This and the above "if" are in the wrong order. If there's an alias we should just use that as the very first thing since there's no other logic that needs to happen.